packages/flutter/lib/src/painting/shader_warm_up.dart

```diff
@@ -0,0 +1,144 @@
+import 'dart:developer';
+import 'dart:ui';
+
+/// Interface for drawing an image to warm up Skia shader compilations.
+///
+/// When Skia first sees a certain type of draw operations on GPU, it needs to
+/// compile the corresponding shader. The compilation can be slow (20ms-200ms).
+/// Having that time as a startup latency is much better than having a jank in
+/// the middle of an animation.
+///
+/// Therefore we use this during the [runApp] call to move common shader
+/// compilations from animation time to startup time. By default, a
+/// [DefaultShaderWarmUp] is used. Create a custom [ShaderWarmUp] subclass to
+/// replace it if needed.
+///
+/// This warm up needs to be run on each individual device because the shader
+/// compilation depends on the specific GPU hardware and driver a device has. It
+/// can't be pre-computed during the Flutter engine compilation as the engine is
+/// device agnostic.
+abstract class ShaderWarmUp {
+  /// Allow const constructors for subclasses.
+  const ShaderWarmUp();
+
+  /// Width of the image to be drawn.
+  int get width;
+
+  /// Height of the image to be drawn.
+  int get height;
```

Comment:
rather than width and height, return a Size.

Also, consider adding commentary explaining what size to return (e.g. "The size should be similar to the size of the device's screen. 1024x1024 is a good choice as it is within an order of magnitude of most devices.").

Why don't we use the actual screen size? I guess we don't know it early enough in release builds?